### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by carmelopellegrino**
These small edits allow reducing the computation time from ~10 seconds to ~2.6 on an Core i5-3330 processor.

### CODE REVIEW

File: multithreading.c [hunk 1]
```diff
@@ -25,7 +25,7 @@ void *thr_func(void *arg) {
   thread_data_t *data = (thread_data_t *)arg;
 
  // making sure that every thread gets a different set of random numbers
-  int seed = time(NULL)+data->tid;
+  int seed = time(NULL) + (1 + data->tid);
```
Comment (valerionew) [inline multithreading.c]: I've already corrected the multiplying by zero error in a previous commit (here https://github.com/5N44P/computational-madness/commit/af902d360f63b6c459a657d9d9eed8db274f8158) by chainging it to a sum

File: multithreading.c [hunk 2]
```diff
@@ -44,9 +45,9 @@ void *thr_func(void *arg) {
 
     //the enhanced random formula
     x=x1*RAND_MAX_P1+x2;
-    x=x/randmax_big;
+    x=x * randmax_big;
```
Comment (valerionew) [inline multithreading.c]: just a small oversight here: should be `irandmax_big` instead of `randmax_big`. Also on line 50

**Comment by valerionew** [comment]
Impressive! 
The computation time on my computer (Core i7-2630QM) went down to one third of the original code time.
I never would've thought that just a few, smart, changes would so drastically change the computation time!

File: multithreading.c [hunk 3]
```diff
@@ -54,6 +54,7 @@ void *thr_func(void *arg) {
     }
   }
 
+  data->reentrantSeed = seed;
```
Comment (valerionew) [inline multithreading.c:58]: Just a reminder for myself: I don't need the seed in the struct, i can remove it from the struct

File: multithreading.c (hunk 1 above)
Comment (carmelopellegrino) [reply multithreading.c]: Ah I see... It was multiplication before :P

File: multithreading.c (hunk 2 above)
Comment (carmelopellegrino) [reply multithreading.c]: Yes, you're right!

**Comment by carmelopellegrino** [comment]
Search for "cache miss". It is something every parallel-computing developer should be aware of :)

**Pull request merged**